frontend/javascripts/libs/toast.tsx

```diff
@@ -112,13 +112,29 @@ const Toast = {
     }
 
     notification[type](toastConfig);
+
+    // Make sure that toasts don't just dissapear while user has WK in background tab (e.g. while uploading large dataset).
+    // Most browsers pause requestAnimationFrame() if the current tab is not active, but Firefox does not seem to do that.
+    if (!sticky && key != null) {
+      requestAnimationFrame(async () => {
```

Comment:
You can use `await animationFrame();` to avoid the callback style. `animationFrame` is a helper in libs/utils.ts.